proto/dalc/dalc.proto

```diff
@@ -0,0 +1,49 @@
+syntax = "proto3";
+package dalc;
+option go_package = "github.com/celestiaorg/optimint/types/pb/dalc";
+
+import "optimint/optimint.proto";
+
+enum StatusCode {
+	STATUS_CODE_UNSPECIFIED = 0;
+	STATUS_CODE_SUCCESS = 1;
+	STATUS_CODE_TIMEOUT = 2;
+	STATUS_CODE_ERROR   = 3;
+}
+
+message DAResponse {
+	StatusCode code = 1;
+	string message = 2;
+}
+
+message SubmitBlockRequest {
+	optimint.Block block = 1;
+}
+
+message SubmitBlockResponse {
+	DAResponse result = 1;
+}
+
+message CheckBlockAvailabilityRequest {
+	optimint.Header header = 1;
+}
+
+message CheckBlockAvailabilityResponse {
+	DAResponse result = 1;
+	bool data_available = 2;
+}
+
+message RetrieveBlockRequest {
+	uint64 height = 1;
+}
+
+message RetrieveBlockResponse {
+	DAResponse result = 1;
+	optimint.Block block = 2;
+}
+
+service DataAvailabilityLayerClientService {
+	rpc SubmitBlock(SubmitBlockRequest) returns (SubmitBlockResponse) {}
+	rpc CheckBlockAvailability(CheckBlockAvailabilityRequest) returns (CheckBlockAvailabilityResponse) {}
+	rpc RetrieveBlock(RetrieveBlockRequest) returns (RetrieveBlockResponse) {}
+}
```

Comment:
Before celestia-node is ready, we need some kind of DALC that can be accessed by multiple clients on multiple machines, for testing.
Additionally, I was thinking about having Celestia DALC over gRPC in final solution. The rationale is following: DALC is only a "pluggable" part of Optimint. Running entire node (celestia or IPFS) in such a "plugin" seems to be a heavyweight approach.

Reply:
Yes, we're on the same page.
celestiaorg/celestia-messages#1 seems to be a good candidate for separate library anyways.

There definitely are some trade-offs. That's why current code base is flexible and allows both "build-in" DALC implementations (like `da/mock`), and gRPC.

If we can have "library style" celestia DALC (with submission, availability check and retrieval) without dependency hell, this would be really awesome.

Another idea is, that all celestia types (and corresponding protobufs) are extracted to a separate module/repo so anyone can use them without any extra dependencies.